specs/WebResourceResponseReceived.md

```diff
@@ -0,0 +1,232 @@
+
+m_webview->add_WebResourceResponseReceived(
+    Callback<ICoreWebView2WebResourceResponseReceivedEventHandler>(
+        [this](ICoreWebView2* webview, ICoreWebView2WebResourceResponseReceivedEventArgs* args)
+            -> HRESULT {
+            // The request object as committed
+            wil::com_ptr<ICoreWebView2WebResourceRequest> webResourceRequest;
+            CHECK_FAILURE(args->get_Request(&webResourceRequest));
+            // The response object as received
+            wil::com_ptr<ICoreWebView2WebResourceResponse> webResourceResponse;
+            CHECK_FAILURE(args->get_Response(&webResourceResponse));
+            
+            // Get body content for the response. Redirect responses will
+            // return an error HRESULT as their body (if any) is ignored.
+            HRESULT populateCallResult = args->PopulateResponseContent(
+                Callback<
+                    ICoreWebView2WebResourceResponseReceivedEventArgsPopulateResponseContentCompletedHandler>(
+                    [this, webResourceRequest, webResourceResponse](HRESULT result) {
+                        // The response might not have a body.
+                        bool populatedBody = SUCCEEDED(result);
+                        
+                        std::wstring message =
+                            L"{ \"kind\": \"event\", \"name\": "
+                            L"\"WebResourceResponseReceived\", \"args\": {"
+                            L"\"request\": " +
+                            RequestToJsonString(webResourceRequest.get()) +
+                            L", "
+                            L"\"response\": " +
+                            ResponseToJsonString(webResourceResponse.get()) + L"}";
+
+                        message +=
+                            WebViewPropertiesToJsonString(m_webview.get());
+                        message += L"}";
+                        PostEventMessage(message);
+                        return S_OK;
+                    })
+                    .Get());
+
+            return S_OK;
+        })
+        .Get(),
+    &m_webResourceResponseReceivedToken);
+```
+
+## C#
+```c#
+WebView.WebResourceResponseReceived += WebView_WebResourceResponseReceived;
+
+// Note: modifications made to request and response are ignored
+private async void WebView_WebResourceResponseReceived(object sender, CoreWebView2WebResourceResponseReceivedEventArgs e)
+{
+    // Actual headers sent with request
+    foreach (var current in e.Request.Headers)
+        Console.WriteLine(current);
+
+    // Headers in response received
+    foreach (var current in e.Response.Headers)
+        Console.WriteLine(current);
+
+    // Status code from response received
+    int status = e.Response.StatusCode;
+    if (status == 200)
+    {
+        // Handle
+        Console.WriteLine("Request succeeded!");
+
+        // Get response body
+        try
+        {
+            await e.PopulateResponseContentAsync();
```

Comment:
Are there any ordering concerns here? The event source is going to carry on while we're awaiting content here.

Reply:
Yes, it will carry on. The continuation for `PopulateResponseAsync` will be executed once the response content has finished loading, which is async.